python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -556,6 +596,33 @@ def filter_out_invalid_p1_times(self, clear_existing: bool = False):
 
         return self
 
+    def get_available_source_ids(self, num_messages_to_read: int = 10):
+        self._populate_available_source_ids(num_messages_to_read=num_messages_to_read)
+
+    def _populate_available_source_ids(self, num_messages_to_read):
+        self.available_source_ids = set()
+        # Loop over all message types and read N of each type.
+        for message_type in np.unique(self.index['type']):
+            try:
+                message_type = MessageType(message_type, raise_on_unrecognized=True)
+            except (KeyError, ValueError) as e:
+                continue
```

Comment:
Instead of this, I think we should allow unrecognized types above and then treat them the same as all others below. Source ID is part of the message header, so there's no need to decode the payload to get it and unsupported types (i.e., types we don't have classes defined for and that we can't decode the payload for) should still count.

You should test this just to be safe to make sure it does actually work when you make this change. Change the pose message enum definition to a wrong value, then process a log and make sure it still handles message 10000 correctly and populates the list of source IDs.

Reply:
Okay this does actually give us an error; the mixed log reader doesn't work when we call `filter_in_place()` with an unsupported message type. In particular, this line gives an error because it's an "unsupported key type":
```
self.index = self.index[key]
```